app/src/main/java/me/saket/dank/ui/accountmanager/AccountManagerActivity.java

```diff
@@ -0,0 +1,325 @@
+import javax.inject.Inject;
+
+import me.saket.dank.R;
+import me.saket.dank.data.ErrorResolver;
+import me.saket.dank.data.ResolvedError;
+import me.saket.dank.di.Dank;
+import me.saket.dank.di.RootComponent;
+import me.saket.dank.ui.DankActivity;
+import me.saket.dank.ui.authentication.LoginActivity;
+import me.saket.dank.ui.user.UserSessionRepository;
+import me.saket.dank.ui.accountmanager.AccountManagerAdapter.AccountManagerViewHolder;
+import me.saket.dank.ui.subscriptions.SubscriptionRepository;
+import me.saket.dank.utils.ItemTouchHelperDragAndDropCallback;
+import me.saket.dank.utils.RxDiffUtil;
+import me.saket.dank.utils.itemanimators.SlideUpAlphaAnimator;
+import me.saket.dank.widgets.swipe.RecyclerSwipeListener;
+
+@DeepLink(AccountManagerActivity.DEEP_LINK)
+@RequiresApi(Build.VERSION_CODES.N_MR1)
+public class AccountManagerActivity extends DankActivity {
+  public static final String DEEP_LINK = "dank://accountManager";
+
+  private final int ACTION_DELETE = 1;
+  private final int ACTION_LOGOUT = 2;
+  private int ACTION = 0;
+
+  private AccountManager selectedAccount = null;
+
+  @BindView(R.id.account_manager_root) ViewGroup rootViewGroup;
+  @BindView(R.id.account_manager_accounts_recyclerview) RecyclerView accountsRecyclerView;
+  @BindView(R.id.account_manager_logout) Button logoutButton;
+  @BindView(R.id.account_progressbar) View loadingProgessbar;
+
+  @Inject Lazy<UserSessionRepository> userSessionRepository;
+  @Inject Lazy<SubscriptionRepository> subscriptionRepository;
+  @Inject Lazy<AccountManagerRepository> accountManagerRepository;
+  @Inject Lazy<AccountManagerAdapter> accountManagerAdapter;
+  @Inject Lazy<ErrorResolver> errorResolver;
+
+  private Disposable confirmTimer = Disposables.disposed();
+  private Disposable timerDisposable = Disposables.empty();
+
+  @CheckResult
+  public static Intent intent(Context context) {
+    return new Intent(context, AccountManagerActivity.class);
+  }
+
+  @Override
+  protected void onCreate(@Nullable Bundle savedInstanceState) {
+    Dank.dependencyInjector().inject(this);
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_account_manager);
+    ButterKnife.bind(this);
+  }
+
+  @Override
+  protected void onPostCreate(@Nullable Bundle savedState) {
+    super.onPostCreate(savedState);
+
+    setupUserList();
+  }
+
+  @Override
+  protected void onSaveInstanceState(Bundle outState) {
+    super.onSaveInstanceState(outState);
+  }
+
+  @Override
+  @OnClick(R.id.account_manager_done)
+  public void finish() {
+    timerDisposable.dispose();
+    super.finish();
+  }
+
+  @OnClick(R.id.account_manager_logout)
+  public void logout() {
+    confirmAction(ACTION_LOGOUT);
+  }
+
+  private Completable queueToDelete(AccountManager account){
+    this.selectedAccount = account;
+    return this.confirmAction(ACTION_DELETE);
+  }
+
+  private Completable confirmAction(int action) {
+    if (confirmTimer.isDisposed()) {
+      ACTION = action;
```

Comment:
Is this necessary? I can see that you are using `ACTION` only inside of this method. You could use the value passed to parameter directly and remove the field.

Reply:
Will check.